python/deltalake/fs.py

```diff
@@ -12,23 +12,47 @@ class DeltaStorageHandler(DeltaFileSystemHandler, FileSystemHandler):
     DeltaStorageHandler is a concrete implementations of a PyArrow FileSystemHandler.
     """
 
-    def open_input_file(self, path: str) -> pa.PythonFile:
+    known_sizes: Dict[str, int] = {}
+
+    def __new__(  # type:ignore
+        cls,
+        table_uri: str,
+        storage_options: Optional[Dict[str, str]] = None,
+        known_sizes: Optional[Dict[str, int]] = None,
+    ):
+        return super().__new__(
+            cls, table_uri=table_uri, options=storage_options  # type:ignore
+        )
+
+    def __init__(
+        self,
+        table_uri: str,
+        storage_options: Optional[Dict[str, str]] = None,
+        known_sizes: Optional[Dict[str, int]] = None,
+    ):
+        if known_sizes:
+            self.known_sizes = known_sizes
+        return
```

Comment:
Could we push the `known_sizes` mapping down into the Rust `DeltaFileSystemHandler`? Then we won't need to change the signatures of `open_input_file` or `open_input_stream`.

Reply:
Good point, fixed this 👍